docs/pai-protocol.yaml

```diff
@@ -0,0 +1,218 @@
+protocolVersion: String, required # Protocol version, current version is 2.
+name: String, required
+type: String, required # The type of the component. Must be one of the following: job, data, script, dockerimage, or output.
+version: String, optional # Component version, default is latest.
+contributor: String, optional
+description: String, optional
+
+prerequisites: # Optional
+  - protocolVersion: String, optional # If omitted, follow the protocolVersion in root.
+    name: String, required
+    type: String, required # Component type. Must be one of the following: data, script, dockerimage, or output. Prerequisites.type cannot be "job".
+    version: String, optional # Component version, Default is latest.
+    contributor: String, optional
+    description: String, optional
+    auth: Object, optional # Only available when the type is dockerimage.
+      username: String, optional
+      passward: String, optional
+      registryuri: String, optional
+    uri: String or list, required # Only when the type is data can the uri be a list.
```

Comment:
Is here a recursive reference of another component defined also by the protocol?